civicrm.user.inc

```diff
@@ -11,6 +11,33 @@ use Drupal\Core\Session\AccountInterface;
  * Implements hook_user_login().
  */
 function civicrm_user_login(AccountInterface $account) {
+  $civicrm = \Drupal::service('civicrm');
+  $civicrm->initialize();
+
+  if (\Drupal::hasService('masquerade')) {
+    // This copies some of the logic from CRM_Core_BAO_UFMatch::synchronize
+    // to avoid a full session reset when masquerading as another user.
+    $session = \CRM_Core_Session::singleton();
+
+    $userID = $session->get('userID');
+    $ufID = $session->get('ufID');
+
+    $userSystem = CRM_Core_Config::singleton()->userSystem;
+    $userSystemID = $userSystem->getBestUFID($user);
```

Comment:
looks like `$user` variable is not defined in this function?

Reply:
:+1: yikes, indeed. Although it produced a notice, but it was still giving the correct behaviour because `getBestUFID` accepts null.

I fixed the code nonetheless and tested to make sure that fixing it did not introduce a regression.